mrtt-ui/src/views/Auth/ResetPasswordForm.js

```diff
@@ -0,0 +1,107 @@
+import { Controller, useForm } from 'react-hook-form'
+import { FormLabel, TextField } from '@mui/material'
+import { toast } from 'react-toastify'
+import { useNavigate, useSearchParams } from 'react-router-dom'
+import { useState } from 'react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import * as yup from 'yup'
+import axios from 'axios'
+
+import { ButtonContainer, RowFlexEnd, ContentWrapper } from '../../styles/containers'
+import { ButtonSubmit } from '../../styles/buttons'
+import { ErrorText, PageTitle } from '../../styles/typography'
+import { Form } from '../../styles/forms'
+import language from '../../language'
+import RequiredIndicator from '../../components/RequiredIndicator'
+
+const pageLanguage = language.pages.resetPassword
+
+const validationSchema = yup.object({
+  password: yup
+    .string()
+    .required(language.form.required)
+    .min(8, pageLanguage.validation.passwordMinimumCharacters),
+  confirmationPassword: yup
+    .string()
+    .required(language.form.required)
+    .oneOf([yup.ref('password')], pageLanguage.validation.passwordsMustMatch)
+})
+
+const formDefaultValues = { password: '', confirmationPassword: '' }
+
+const passwordUrl = `${process.env.REACT_APP_AUTH_URL}/users/password`
+
+const ResetPasswordForm = () => {
+  const [searchParams] = useSearchParams()
+  const [isSubmitError, setIsSubmitError] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const navigate = useNavigate()
+
+  const {
+    control: formControl,
+    handleSubmit: validateInputs,
+    formState: { errors }
+  } = useForm({ resolver: yupResolver(validationSchema), defaultValues: formDefaultValues })
+
+  const handleSubmit = ({ password, confirmationPassword }) => {
+    axios
+      .put(passwordUrl, {
+        user: {
+          password,
+          password_confirmation: confirmationPassword,
+          reset_password_token: searchParams.get('token')
+        }
+      })
+      .then((response) => {
+        if (response.data.message === 'The password was reset successfully') {
+          setIsSubmitting(false)
+          toast.success(pageLanguage.success)
+          navigate('/auth/login')
+        } else {
+          throw new Error('This didnt work. Your token probably expired.')
+        }
+      })
```

Comment:
So this is to address a weird  api quirk with out too much back and forth.  When the token has expired or is a bad token, the api still gives a 200. So I check the message that gets sent on a _real_ success, and trigger the promise to reject otherwise (catch runs)